lib/temporal/workflow/errors.rb

```diff
@@ -0,0 +1,39 @@
+module Temporal
+  class Workflow
+    class Errors
+      include Concerns::Payloads
+
+      # Convert a failure returned from the server to an Error to raise to the client
+      # failure: Temporal::Api::Failure::V1::Failure
+      def error_from(failure, default_exception_class = StandardError)
+        case failure.failure_info
```

Comment:
Copied verbatim from the state manager.  This could be a good candidate for unit tests if you want to beef up coverage and correctness here.